examples/stitches/app/root.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  Links,
+  LiveReload,
+  Meta,
+  Outlet,
+  Scripts,
+  ScrollRestoration,
+  useCatch
+} from "remix";
+import { useContext, useEffect } from "react";
+import ServerStyleContext from "./styles/server.context";
+import ClientStyleContext from "./styles/client.context";
+import type { MetaFunction } from "remix";
+
+import { styled } from "./styles/stitches.config";
+
+const Container = styled("div", {
+  backgroundColor: "#ff0000",
+  padding: "1em"
+});
+
+export const meta: MetaFunction = () => {
+  return { title: "Remix with Stitches" };
+};
+
+interface DocumentProps {
+  children: React.ReactNode;
+  title?: string;
+}
+
+const Document = ({ children, title }: DocumentProps) => {
+  const serverStyleData = useContext(ServerStyleContext);
+  const clientStyleData = useContext(ClientStyleContext);
+
+  // Only executed on client
+  useEffect(() => {
+    // reset cache to re-apply global styles
+    clientStyleData.reset();
+  }, [clientStyleData, serverStyleData]);
+
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width,initial-scale=1" />
+        {title ? <title>{title}</title> : null}
+        <Meta />
+        <Links />
+        <style
```

Comment:
@helderburato should this be something like:

```jsx
dangerouslySetInnerHTML={{
						__html: clientStyleData.sheet || serverStyleData,
					}}
```

I noticed with the setup as is I was getting unstyled SSR on initial load, but this seems to do the trick.